oneandone/src/test/java/org/apache/jclouds/oneandone/rest/features/SharedStorageApiLiveTest.java

```diff
@@ -0,0 +1,152 @@
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+@Test(groups = "live", testName = "SharedStorageApiLiveTest")
+public class SharedStorageApiLiveTest extends BaseOneAndOneLiveTest {
+
+   private SharedStorage currentSharedStorage;
+   private Server currentServer;
+   private List<SharedStorage> sharedStorages;
+
+   private SharedStorageApi sharedStorageApi() {
+
+      return api.sharedStorageApi();
+   }
+
+   @BeforeClass
+   public void setupTest() {
+      currentServer = createServer("sharestorage jclouds server");
+      assertNodeAvailable(currentServer);
+      currentSharedStorage = sharedStorageApi().create(SharedStorage.CreateSharedStorage.builder()
+              .name("jcloudsStorage")
+              .description("desc")
+              .size(50)
+              .build());
+      assertNotNull(currentSharedStorage);
+   }
+
+   @Test
+   public void testList() {
+      sharedStorages = sharedStorageApi().list();
+
+      assertNotNull(sharedStorages);
+      Assert.assertTrue(sharedStorages.size() > 0);
+   }
+
+   public void testListWithOption() {
+      GenericQueryOptions options = new GenericQueryOptions();
+      options.options(0, 0, null, "jcloudsStorage", null);
+      List<SharedStorage> imageWithQuery = sharedStorageApi().list(options);
+
+      assertNotNull(imageWithQuery);
+      Assert.assertTrue(imageWithQuery.size() > 0);
+   }
+
+   public void testGet() {
+      SharedStorage result = sharedStorageApi().get(currentSharedStorage.id());
+
+      assertNotNull(result);
+      assertEquals(result.id(), currentSharedStorage.id());
+   }
+
+   @Test(dependsOnMethods = "testGet")
+   public void testUpdate() throws InterruptedException {
+      String updatedName = "Updatedjava";
+
+      SharedStorage updateResult = sharedStorageApi().update(currentSharedStorage.id(), SharedStorage.UpdateSharedStorage.create(updatedName, "desc", null));
+
+      assertNotNull(updateResult);
+      assertEquals(updateResult.name(), updatedName);
+
+   }
+
+   @Test(dependsOnMethods = "testUpdate")
+   public void testAttachServer() throws InterruptedException {
+
+      List<SharedStorage.Server.CreateServer.ServerPayload> servers = new ArrayList<SharedStorage.Server.CreateServer.ServerPayload>();
+      SharedStorage.Server.CreateServer.ServerPayload toAdd = SharedStorage.Server.CreateServer.ServerPayload.create(currentServer.id(), Types.StorageServerRights.R);
+      servers.add(toAdd);
+      SharedStorage updateResult = sharedStorageApi().attachServer(currentSharedStorage.id(), SharedStorage.Server.CreateServer.create(servers));
+
+      assertNotNull(updateResult);
+
+   }
+
+   @Test(dependsOnMethods = "testAttachServer")
+   public void testListServers() {
+      List<SharedStorage.Server> servers = sharedStorageApi().getServers(currentSharedStorage.id());
+
+      assertNotNull(servers);
+      Assert.assertTrue(servers.size() > 0);
+   }
+
+   @Test(dependsOnMethods = "testListServers")
```

Comment:
Better depend on the attach test too, to minimize the chances for this to be skipped.